armada/utils/source.py

```diff
@@ -0,0 +1,65 @@
+import pygit2
+import requests
+import tarfile
+import tempfile
+import shutil
+from os import path
+
+from ..exceptions import source_exceptions
+
+def git_clone(repo_url, branch='master'):
+    '''
+    clones repo to a /tmp/ dir
+    '''
+
+    if repo_url == '':
+        raise source_exceptions.GitLocationException(repo_url)
+
+    _tmp_dir = tempfile.mkdtemp(prefix='armada', dir='/tmp')
+
+    try:
+        pygit2.clone_repository(repo_url, _tmp_dir, checkout_branch=branch)
+    except Exception:
+        raise source_exceptions.GitLocationException(repo_url)
+
+    return _tmp_dir
+
+def get_tarball(tarball_url):
+    tarball_path = download_tarball(tarball_url)
+    return extract_tarball(tarball_path)
+
+def download_tarball(tarball_url):
+    '''
```

Comment:
they can be. for example, in tar_example.yaml when the mariadb-0.1.0.tgz is extracted, the first directory level inside the tarball is 'mariadb' so the yaml has the subpath as 'mariadb'